ksqldb-engine/src/main/java/io/confluent/ksql/function/udf/string/RPad.java

```diff
@@ -49,4 +52,41 @@ public String rpad(
     sb.setLength(targetLen);
     return sb.toString();
   }
+
+  @Udf
+  public ByteBuffer rpad(
+      @UdfParameter(description = "Bytes to be padded") final ByteBuffer input,
+      @UdfParameter(description = "Target length") final Integer targetLen,
+      @UdfParameter(description = "Padding bytes") final ByteBuffer padding) {
+
+    if (input == null) {
+      return null;
+    }
+    if (padding == null
+        || BytesUtils.getByteArray(padding).length == 0
+        || targetLen == null
+        || targetLen < 0) {
+      return null;
+    }
+
+    final byte[] start = BytesUtils.getByteArray(input);
```

Comment:
I think working with a `byte[]` is a lot more readable than trying to work with a buffer here, do others agree/disagree?

Reply:
Yes. I'm actually working in a PR to move away from ByteBuffer because some issues with internal positions of it.